# Projecte final

## Conjunt de dades escollit

El siguiente conjunto de datos se ha extraido de la web del ayuntamiento de Barrcelona https://opendata-ajuntament.barcelona.cat/es

## Ficheros descargados:

### 1) Nuevo fichero General de Entidades Ciudadanas de la ciudad de Barcelona (12/04/2023)
El Fichero General de Entidades Ciudadanas es uno de los instrumentos de relación del Ayuntamiento de Barcelona con las entidades y grupos de hecho de la ciudad. Su creación está prevista en las Normas reguladoras de la participación ciudadana (artículo 9) y está reglamentado mediante las Normas reguladoras del Fichero general de entidades ciudadanas (BOP 26/2/2004).

- Gerencia: Gerència de Drets de Ciutadania, Participació i Transparència  
- Departamento: Departament d'Associacionisme i Iniciatives Ciutadanes  
- Fuente: Comissionat de Participació Ciutadana i Associacionisme  
- Agenda 2030. ODS Principal: ODS 16: Paz, justícia e instituciones sólidas  
- Agenda 2030. ODS Colateral 1: ODS 17: Alianza para lograr los objetivos  
- Frecuencia actualización: Mensual  
- Información histórica: Sí

#### Explicación de las variables del fichero:  

01.**Tipologia_E_GF**: Tipologia de l'entitat. Indica si es tracta d'una entitat o grup de fet. Totes les entitats tenen aquest camp amb el valor "E"  
02.**Nif**: NIF de l'entitat  
03.**Total_Voluntaris**: Nombre de voluntaris que participen a les activitats de l'entitat  
04.**Total_Professionals**: Nombre total de professionals que treballen a l'entitat  
05.**Tipus_Activitat**: Indica si l'activitat és de tipus "permanent" o "temporal"  
06.**Forma_Juridica**: Indica la forma jurídica amb la qual s'ha constituït l'entitat  
07.**Codi_E_GF**: Codi identificador de l'entitat  
08.**Nom_E_GF**: Nom de l'entitat  
09.**Nom_Popular**: Nom pel qual es coneix popularment a l'entitat  
10.**Sigla**: Sigla corresponent a l'entitat  
11.**Nom_Poblacio**: Nom de la població on es troba l'entitat  
12.**Codi_Districte**: Codi de districte seguint la codificació de l'Ajuntament de Barcelona  
13.**Nom_Districte**: Nom del districte  
14.**Codi_Barri**: Codi del barri  
15.**Nom_Barri**: Nom del barri  
16.**Url_Preferent**: URL preferent  
17.**Any_Constitucio**: Any de constitució de l'entitat  
18.**Sector_Territorial**: Territori en el qual l'entitat desenvolupa les seves activitats  
19.**Total_Socis**: Nombre total de socis de l'entitat  
20.**Telefon_Preferent**: Telèfon preferent de contacte  
21.**Correu_Preferent**: Adreça preferent de correu electrònic  
22.**Sector_Tematic_1**: Primer sector temàtic  
23.**Sector_Tematic_2**: Segon sector temàtic  
24.**Sector_Tematic_3**: Tercer sector temàtic  
25.**XXSS_Instagram**: Instagram  
26.**XXSS_Twitter**: Twitter  
27.**XXSS_Facebook**: Facebook  
28.**XXSS_Altres**: Altres xarxes socials  
29.**Codi_tipus_via**: Codi tipus via codificat AjBCN  
30.**Tipus_Via**: Tipus via  
31.**Codi_Carrer**: Codi carrer codificació AjBCN  
32.**Nom_Carrer**: Nom del carrer  
33.**Codi_número_carrer**: Codi número carrer codificació AjBCN  
34.**Numero_Carrer**: Número  
35.**Escala**: Escala  
36.**Pis**: Pis  
37.**Porta**: Porta   
38.**Latitud**: Latitud     
39.**Longitud**: Longitud   
40.**X_ETRS89**: Coordenada X en ETRS89  
41.**Y_ETRS89**: Coordenada Y en ETRS89

### 2) Subvenciones concedidas a entitades de la ciudad de Barcelona (12/04/2023)
Este conjunto de datos incluye las subvenciones otorgadas a las diferentes entidades inscritas en el Fichero General de Entidades de la ciudad de Barcelona, incluyendo el título del proyecto e importe de la subvención.

- Gerencia: Gerència de Drets de Ciutadania, Participació i Transparència  
- Departamento: Departament d'Associacionisme i Iniciatives Ciutadanes  
- Fuente: Ajuntament de Barcelona  
- Agenda 2030. ODS Principal: ODS 16: Paz, justícia e instituciones sólidas  
- Agenda 2030. ODS Colateral 1: ODS 17: Alianza para lograr los objetivos  
- Frecuencia actualización: Anual  
- Información histórica: No

#### Explicación de las variables del fichero:  

01.**Tipologia_E_GF**: Tipologia de l'entitat.Indica si es tracta d'una entitat o grup de fet. Totes les entitats tenen aquest camp amb el valor "E"  
02.**Codi_E_GF**: Codi intern de l'entitat. Consultar Nou fitxer General d'Entitats Ciutadanes de la ciutat de Barcelona  
03.**Entitat_Municipal**: Entitat que atorga la subvenció  
04.**Organ_Gestor**: Òrgan gestor de la subvenció  
05.**Any**: Any de convocatòria de la subvenció  
06.**Tipologia_De_Subvencio**: Tipologia de la subvenció  
07.**Objecte**: Títol del projecte presentat per l'entitat  
08.**Import_Sollicitat**: Import sol·licitat (€)  
09.**Import_Total_Projecte**: Import total del projecte presentat (€)  
10.**Import_Atorgat_Inicial**: Import atorgat inicialment (€)  
11.**Import_Reintegrat_Total**: Import reintegrat (€)

Objetivos:
1) Analizar los datos del fichero general estableciendo la distribución estadística de tipologías de entidades ciudadanas, su distribución por la ciudad, los valores medios de integrantes y socios, el alcance territorial, y el nivel de digitalización.
2) Establecer una analisis de correlaciones entre las subvenciones y las características de las entidades identificando marcadores de interes.

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

## 1) Importación de ficheros, sanity checks y "mergeado" de las bases de datos

### a.  Entidades

In [2]:
df_entidades = pd.read_csv("C:/Users/alpid/Downloads/2023_1S_ENTITIES.csv")
df_entidades.head()

Tipologia_E_GF        Nif  Total_Voluntaris  Total_Professionals  \
0              E  G66024894               0.0                  0.0   
1              E  R0800530H               0.0                  0.0   
2              E  G08489528               0.0                  0.0   
3              E  G01675180               0.0                  0.0   
4              E  F67141648               0.0                  0.0   

  Tipus_Activitat Forma_Juridica    Codi_E_GF  \
0             NaN     Associació  99888005242   
1             NaN     Associació  99746315497   
2             NaN       Fundació  99739953046   
3             NaN     Associació  99674314027   
4             NaN    Cooperativa  99564110147   

                                    Nom_E_GF Nom_Popular Sigla  ...  \
0  Associació d'Afectats d'Urticària Crònica         NaN  AAUC  ...   
1           Parròquia Sant Ignasi de Loiola          NaN   NaN  ...   
2        Fundació Privada Hospital Evangèlic         NaN   NaN  ...   
3     Associació d'Entreteniment Grupo Dance         NaN   NaN  ...   
4                              Ars Viva SCCL         NaN   NaN  ...   

  Nom_Carrer  Codi_número_carrer Numero_Carrer  Escala    Pis Porta  \
0   Guadiana   15120000032000380            32     NaN      1     3   
1   Provença   26800300542005440           544     NaN    NaN   NaN   
2   CAMELIES   05650400013000170       13 - 17     NaN    NaN   NaN   
3       Lugo   18180100046000480            46     NaN  local     2   
4    Augusta   02800000108001100           108     NaN    NaN   NaN   

     Latitud  Longitud    X_ETRS89     Y_ETRS89  
0  41.373245  2.139609  428049.153  4580549.246  
1  41.408154  2.179615  431431.438  4584392.344  
2  41.411072  2.158888  429702.198  4584732.912  
3  41.423155  2.155948  429469.517  4586076.796  
4  41.401329  2.148809  428849.207  4583659.423  

[5 rows x 41 columns]

**Voy a reemplazar los valores 0.0 por Nan para no tener resutlados erroneos**.

In [3]:
df_entidades = df_entidades.replace(0.0, np.nan)
df_entidades.head()

Tipologia_E_GF        Nif  Total_Voluntaris  Total_Professionals  \
0              E  G66024894               NaN                  NaN   
1              E  R0800530H               NaN                  NaN   
2              E  G08489528               NaN                  NaN   
3              E  G01675180               NaN                  NaN   
4              E  F67141648               NaN                  NaN   

  Tipus_Activitat Forma_Juridica    Codi_E_GF  \
0             NaN     Associació  99888005242   
1             NaN     Associació  99746315497   
2             NaN       Fundació  99739953046   
3             NaN     Associació  99674314027   
4             NaN    Cooperativa  99564110147   

                                    Nom_E_GF Nom_Popular Sigla  ...  \
0  Associació d'Afectats d'Urticària Crònica         NaN  AAUC  ...   
1           Parròquia Sant Ignasi de Loiola          NaN   NaN  ...   
2        Fundació Privada Hospital Evangèlic         NaN   NaN  ...   
3     Associació d'Entreteniment Grupo Dance         NaN   NaN  ...   
4                              Ars Viva SCCL         NaN   NaN  ...   

  Nom_Carrer  Codi_número_carrer Numero_Carrer  Escala    Pis Porta  \
0   Guadiana   15120000032000380            32     NaN      1     3   
1   Provença   26800300542005440           544     NaN    NaN   NaN   
2   CAMELIES   05650400013000170       13 - 17     NaN    NaN   NaN   
3       Lugo   18180100046000480            46     NaN  local     2   
4    Augusta   02800000108001100           108     NaN    NaN   NaN   

     Latitud  Longitud    X_ETRS89     Y_ETRS89  
0  41.373245  2.139609  428049.153  4580549.246  
1  41.408154  2.179615  431431.438  4584392.344  
2  41.411072  2.158888  429702.198  4584732.912  
3  41.423155  2.155948  429469.517  4586076.796  
4  41.401329  2.148809  428849.207  4583659.423  

[5 rows x 41 columns]

In [4]:
df_entidades.shape

(5484, 41)

In [6]:
df_entidades.isna().all()

Tipologia_E_GF         False
Nif                    False
Total_Voluntaris       False
Total_Professionals    False
Tipus_Activitat        False
Forma_Juridica         False
Codi_E_GF              False
Nom_E_GF               False
Nom_Popular            False
Sigla                  False
Nom_Poblacio           False
Codi_Districte         False
Nom_Districte          False
Codi_Barri             False
Nom_Barri              False
Url_Preferent          False
Any_Constitucio        False
Sector_Territorial     False
Total_Socis            False
Telefon_Preferent      False
Correu_Preferent       False
Sector_Tematic_1       False
Sector_Tematic_2       False
Sector_Tematic_3       False
XXSS_Instagram         False
XXSS_Twitter           False
XXSS_Facebook          False
XXSS_Altres            False
Codi_tipus_via         False
Tipus_Via              False
Codi_Carrer            False
Nom_Carrer             False
Codi_número_carrer     False
Numero_Carrer          False
Escala        

In [5]:
df_entidades.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5484 entries, 0 to 5483
Data columns (total 41 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Tipologia_E_GF       5484 non-null   object 
 1   Nif                  5481 non-null   object 
 2   Total_Voluntaris     10 non-null     float64
 3   Total_Professionals  9 non-null      float64
 4   Tipus_Activitat      3 non-null      object 
 5   Forma_Juridica       5481 non-null   object 
 6   Codi_E_GF            5484 non-null   int64  
 7   Nom_E_GF             5484 non-null   object 
 8   Nom_Popular          11 non-null     object 
 9   Sigla                988 non-null    object 
 10  Nom_Poblacio         5484 non-null   object 
 11  Codi_Districte       5484 non-null   int64  
 12  Nom_Districte        5484 non-null   object 
 13  Codi_Barri           5484 non-null   int64  
 14  Nom_Barri            5484 non-null   object 
 15  Url_Preferent        2938 non-null   o

In [7]:
df_entidades.isna().sum()

Tipologia_E_GF            0
Nif                       3
Total_Voluntaris       5474
Total_Professionals    5475
Tipus_Activitat        5481
Forma_Juridica            3
Codi_E_GF                 0
Nom_E_GF                  0
Nom_Popular            5473
Sigla                  4496
Nom_Poblacio              0
Codi_Districte            0
Nom_Districte             0
Codi_Barri                0
Nom_Barri                 0
Url_Preferent          2546
Any_Constitucio           0
Sector_Territorial     1510
Total_Socis            4148
Telefon_Preferent       790
Correu_Preferent        603
Sector_Tematic_1          1
Sector_Tematic_2       3894
Sector_Tematic_3       4829
XXSS_Instagram         5159
XXSS_Twitter           4708
XXSS_Facebook          4385
XXSS_Altres            5193
Codi_tipus_via            2
Tipus_Via                 2
Codi_Carrer               2
Nom_Carrer                2
Codi_número_carrer        2
Numero_Carrer             2
Escala                 5252
Pis                 

**Observamos que de 5484 entidades**:  

**En positivo**:  
- Solo 3 no tienen:
    * NIF
    * Forma Jurídica Indetificada
- Solo 1 no tiene indetificado el Sector temático 1
- Solo 2 no tienen forma de localizarse físicamente al faltar:
    * Codigo Tipo de Via, 
    * Tipo de via, Codigo de Calle,
    * Nombre de la calle,
    * Código de número de calle,
    * Número de calle
    * Latitud
    * Longitud
    * X_ETRS89
    * Y_ETRS89  

Verificaremos en cada caso, cuales son, si han sifo subvencionadas y el valor de dichas subvenciones frente al total en ese caso, o si por el contrario podemos deducir que en algún caso la falta de datos puede haber influido para no conseguirla. Verificaremos si ademas tienen telefono o correo o no, o si en lagún caso no hay forma de localizarlas.

**En negativo**:
- Solo:
    * 10 tienen el total de voluntarios indetificado 
    * 9 tienen el total de profesionales indetificado 
    * 3 tienen el tipo de actividad indetificada 
    * 11 tienen nombre popular indetificado
    * 1336 tienen el número total de socios indetificados (24%)
    * 1590 tienen el sector temático 2 identificado (29%)
    * 655 tienen el sector temático 3 identificado (12%)  

Verificaremos si hay correlación entre la existencia de estos parámetros y la asignación de subvenciones.
    
**Por otro lado**:
- 988 tienen Siglas (18%)
- 2938 tienen URL prefernet identificada (53%)
- 3974 tienen Sector territorial idnetificado (72%)
- 4694 tienen teléfono preferente identificado (84%)
- 4881 tienen correo preferente identificado (89%)
- A nivel de presencia en redes solo:
    * 325 tienen Instagram (6%)
    * 776 tienen Twitter (14%)
    * 1099 tienen Facebook (20%)
    * 291 estan en otras redes (5%)
- A nivel de localización solo:
    * 232 tienen escalera identificada (4%)
    * 2813 tienen el Piso idnetificado (51%)
    * 1629 tienen la Puerta indetificada (3%)  

Verificaremos si hay correlación entre la existencia de estos paramentros y las subvenciones. 
 
**En el caso de los siguientes parámetros, todas las entidades los tienen identificados**:
- E_GF:
    * Tipologia_E_GF
    * Codi_E_GF
    * Nom_E_GF
- Localización general:
    * Nombre de Población
    * Código de Distrito
    * Nombre de Distrito
    * Código de Barrio
    * Nombre de Barrio
- Año de constitución  

Verificaremos la distribución de entidades y subvenciones por Distrito o Barrio y el año de cosntitución.


In [8]:
df_entidades.describe()

Total_Voluntaris  Total_Professionals     Codi_E_GF  Codi_Districte  \
count         10.000000             9.000000  5.484000e+03     5484.000000   
mean          27.200000            28.444444  8.379581e+10        4.885120   
std           14.289079            56.409465  2.916941e+10        3.113443   
min            4.000000             1.000000  8.600000e+01        1.000000   
25%           16.250000             3.000000  9.208600e+10        2.000000   
50%           31.000000             4.000000  9.940003e+10        5.000000   
75%           35.000000             7.000000  9.940026e+10        8.000000   
max           50.000000           170.000000  9.988801e+10       10.000000   

        Codi_Barri  Any_Constitucio    Total_Socis  Codi_tipus_via  \
count  5484.000000      5484.000000    1336.000000     5482.000000   
mean     27.624726      1826.541758    1119.911677        4.429040   
std      22.699922       199.951798   11229.099830        6.730278   
min       1.000000      1380.000000       2.000000        2.000000   
25%       7.000000      1600.000000      15.000000        2.000000   
50%      22.000000      1987.000000      65.000000        2.000000   
75%      44.000000      2007.000000     250.000000        2.000000   
max      73.000000      2023.000000  300000.000000       89.000000   

         Codi_Carrer      Latitud     Longitud       X_ETRS89      Y_ETRS89  
count    5482.000000  5482.000000  5482.000000    5482.000000  5.482000e+03  
mean   206165.331084    41.400057     2.164804  430184.608608  4.583506e+06  
std    146065.703179     0.021541     0.022638    1900.782356  2.385003e+03  
min       102.000000    41.324199     2.080983  423204.733000  4.575106e+06  
25%     94300.000000    41.382874     2.149748  428927.027250  4.581598e+06  
50%    190303.000000    41.397052     2.166016  430284.951000  4.583169e+06  
75%    287100.750000    41.414350     2.179538  431425.966500  4.585073e+06  
max    701719.000000    41.463847     2.221681  434950.268000  4.590571e+06

### b. Subvenciones

In [9]:
df_subvenciones = pd.read_csv("C:/Users/alpid/Downloads/SUBVENCIONS.csv")
df_subvenciones.head()

Tipologia_E_GF    Codi_E_GF                     Entitat_Municipal  \
0              E          224               Ajuntament de Barcelona   
1              E  99400147070  Institut Municipal Barcelona Esports   
2              E          224               Ajuntament de Barcelona   
3              E  99400173016  Institut Municipal Barcelona Esports   
4              E          224               Ajuntament de Barcelona   

                                        Organ_Gestor   Any  \
0   Gerència de Política Econòmica i Desenvolupament  2019   
1                                                NaN  2017   
2   Gerència de Política Econòmica i Desenvolupament  2019   
3                                                NaN  2017   
4  Gerència d’Àrea d’Economia, Recursos i Promoci...  2022   

                              Tipologia_De_Subvencio  \
0                            SUBVENCIONS NOMINATIVES   
1         SUBVENCIONS DIRECTES (amb o sense CONVENI)   
2  Convocatòria Subvencions Promoció i Reforç Eco...   
3         SUBVENCIONS DIRECTES (amb o sense CONVENI)   
4  Convocatòria 2022 Impuls socioeconòmic del ter...   

                                       Objecte  Import_Sollicitat  \
0  POSADA EN MARXA DEL PROJECTE SOCIAL BICIHUB           120000.0   
1                Ajuts 2017-2018 (1r pagament)                0.0   
2                        Comuniquem el BiciHub             8000.0   
3                Ajuts 2017-2018 (1r pagament)                0.0   
4             L'estiu Cooperatiu de Sant Martí            40000.0   

   Import_Total_Projecte  Import_Atorgat_Inicial  Import_Reintegrat_Total  
0              120000.00               120000.00                      0.0  
1                1377.00                 1377.00                      0.0  
2               10000.00                 4000.00                      0.0  
3                7917.75                 7917.75                      0.0  
4               50000.00                18000.00                      0.0

**Igual que con df_entidades, voy a convertir todos los 0.0 en Nan para evitar errores:**

In [10]:
df_subvenciones = df_subvenciones.replace(0.0, np.nan)
df_subvenciones.head()

Tipologia_E_GF    Codi_E_GF                     Entitat_Municipal  \
0              E          224               Ajuntament de Barcelona   
1              E  99400147070  Institut Municipal Barcelona Esports   
2              E          224               Ajuntament de Barcelona   
3              E  99400173016  Institut Municipal Barcelona Esports   
4              E          224               Ajuntament de Barcelona   

                                        Organ_Gestor   Any  \
0   Gerència de Política Econòmica i Desenvolupament  2019   
1                                                NaN  2017   
2   Gerència de Política Econòmica i Desenvolupament  2019   
3                                                NaN  2017   
4  Gerència d’Àrea d’Economia, Recursos i Promoci...  2022   

                              Tipologia_De_Subvencio  \
0                            SUBVENCIONS NOMINATIVES   
1         SUBVENCIONS DIRECTES (amb o sense CONVENI)   
2  Convocatòria Subvencions Promoció i Reforç Eco...   
3         SUBVENCIONS DIRECTES (amb o sense CONVENI)   
4  Convocatòria 2022 Impuls socioeconòmic del ter...   

                                       Objecte  Import_Sollicitat  \
0  POSADA EN MARXA DEL PROJECTE SOCIAL BICIHUB           120000.0   
1                Ajuts 2017-2018 (1r pagament)                NaN   
2                        Comuniquem el BiciHub             8000.0   
3                Ajuts 2017-2018 (1r pagament)                NaN   
4             L'estiu Cooperatiu de Sant Martí            40000.0   

   Import_Total_Projecte  Import_Atorgat_Inicial  Import_Reintegrat_Total  
0              120000.00               120000.00                      NaN  
1                1377.00                 1377.00                      NaN  
2               10000.00                 4000.00                      NaN  
3                7917.75                 7917.75                      NaN  
4               50000.00                18000.00                      NaN

In [11]:
df_subvenciones.shape

(56917, 11)

In [12]:
df_subvenciones.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 56917 entries, 0 to 56916
Data columns (total 11 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Tipologia_E_GF           56917 non-null  object 
 1   Codi_E_GF                56917 non-null  int64  
 2   Entitat_Municipal        56910 non-null  object 
 3   Organ_Gestor             51654 non-null  object 
 4   Any                      56917 non-null  int64  
 5   Tipologia_De_Subvencio   56917 non-null  object 
 6   Objecte                  56917 non-null  object 
 7   Import_Sollicitat        55437 non-null  float64
 8   Import_Total_Projecte    56558 non-null  float64
 9   Import_Atorgat_Inicial   56917 non-null  float64
 10  Import_Reintegrat_Total  1485 non-null   float64
dtypes: float64(4), int64(2), object(5)
memory usage: 4.8+ MB


In [13]:
df_subvenciones.isna().all()

Tipologia_E_GF             False
Codi_E_GF                  False
Entitat_Municipal          False
Organ_Gestor               False
Any                        False
Tipologia_De_Subvencio     False
Objecte                    False
Import_Sollicitat          False
Import_Total_Projecte      False
Import_Atorgat_Inicial     False
Import_Reintegrat_Total    False
dtype: bool

In [14]:
df_subvenciones.isna().sum()

Tipologia_E_GF                 0
Codi_E_GF                      0
Entitat_Municipal              7
Organ_Gestor                5263
Any                            0
Tipologia_De_Subvencio         0
Objecte                        0
Import_Sollicitat           1480
Import_Total_Projecte        359
Import_Atorgat_Inicial         0
Import_Reintegrat_Total    55432
dtype: int64

Vemos que en general todas las entidades subveniconadas tienen todos los datos.

In [15]:
df_subvenciones.describe()

Codi_E_GF           Any  Import_Sollicitat  Import_Total_Projecte  \
count  5.691700e+04  56917.000000       5.543700e+04           5.655800e+04   
mean   8.506857e+10   2017.523323       1.338104e+04           2.983565e+04   
std    2.838212e+10      3.110568       7.219469e+04           1.129866e+05   
min    2.240000e+02   2011.000000       6.000000e-01           6.000000e-01   
25%    9.208601e+10   2015.000000       2.000000e+03           4.350000e+03   
50%    9.631921e+10   2018.000000       4.500000e+03           1.025000e+04   
75%    9.940024e+10   2020.000000       1.000000e+04           2.564520e+04   
max    9.988801e+10   2022.000000       7.697740e+06           1.224850e+07   

       Import_Atorgat_Inicial  Import_Reintegrat_Total  
count            5.691700e+04              1485.000000  
mean             1.222339e+04              3031.940343  
std              1.114916e+05              9078.194119  
min             -2.331960e+05                 0.100000  
25%              1.000000e+03               500.000000  
50%              2.000000e+03              1000.000000  
75%              5.400000e+03              2500.000000  
max              7.958646e+06            156886.000000

**Eliminamos columnas del df_entidades que no voy a ultilizar (del df_subvenciones utilizaré, a priori, todas)**:

20.**Telefon_Preferent**: Telèfon preferent de contacte  
21.**Correu_Preferent**: Adreça preferent de correu electrònic  
29.**Codi_tipus_via**: Codi tipus via codificat AjBCN  
30.**Tipus_Via**: Tipus via  
31.**Codi_Carrer**: Codi carrer codificació AjBCN  
32.**Nom_Carrer**: Nom del carrer  
33.**Codi_número_carrer**: Codi número carrer codificació AjBCN  
34.**Numero_Carrer**: Número  
35.**Escala**: Escala  
36.**Pis**: Pis  
37.**Porta**: Porta 38.Latitud: Latitud  
39.**Longitud**: Longitud  
40.**X_ETRS89**: Coordenada X en ETRS89  
41.**Y_ETRS89**: Coordenada Y en ETRS89  

In [16]:
df_entidades.drop(columns=["Sigla", "Telefon_Preferent", "Correu_Preferent", "Codi_tipus_via", "Tipus_Via", "Codi_Carrer", "Nom_Carrer", "Codi_número_carrer", "Numero_Carrer", "Escala", "Pis", "Porta", "Latitud", "Longitud", "X_ETRS89", "Y_ETRS89"], inplace=True)
df_entidades.head()   #He añadido inplace=True para que me escriba directamente en el mismo df.

Tipologia_E_GF        Nif  Total_Voluntaris  Total_Professionals  \
0              E  G66024894               NaN                  NaN   
1              E  R0800530H               NaN                  NaN   
2              E  G08489528               NaN                  NaN   
3              E  G01675180               NaN                  NaN   
4              E  F67141648               NaN                  NaN   

  Tipus_Activitat Forma_Juridica    Codi_E_GF  \
0             NaN     Associació  99888005242   
1             NaN     Associació  99746315497   
2             NaN       Fundació  99739953046   
3             NaN     Associació  99674314027   
4             NaN    Cooperativa  99564110147   

                                    Nom_E_GF Nom_Popular Nom_Poblacio  ...  \
0  Associació d'Afectats d'Urticària Crònica         NaN    BARCELONA  ...   
1           Parròquia Sant Ignasi de Loiola          NaN    BARCELONA  ...   
2        Fundació Privada Hospital Evangèlic         NaN    BARCELONA  ...   
3     Associació d'Entreteniment Grupo Dance         NaN    BARCELONA  ...   
4                              Ars Viva SCCL         NaN    BARCELONA  ...   

   Any_Constitucio Sector_Territorial  Total_Socis     Sector_Tematic_1  \
0             2012          barcelona          NaN                Salut   
1             1600          barcelona          NaN           Religioses   
2             1909          barcelona          NaN                Salut   
3             2020                NaN        110.0              Veïnals   
4             2018          barcelona          NaN  Educació i formació   

                   Sector_Tematic_2  Sector_Tematic_3  \
0                               NaN               NaN   
1                               NaN               NaN   
2                               NaN               NaN   
3  Promoció de la ciutadania activa     Suport social   
4                               NaN               NaN   

                          XXSS_Instagram                   XXSS_Twitter  \
0                                    NaN                            NaN   
1                                    NaN                            NaN   
2                                    NaN  https://twitter.com/NouHE_BCN   
3                                    NaN                            NaN   
4  https://www.instagram.com/arsvivabcn/                            NaN   

                                   XXSS_Facebook XXSS_Altres  
0                                            NaN         NaN  
1                                            NaN         NaN  
2  https://www.facebook.com/Nouhospitalevangelic         NaN  
3                                            NaN         NaN  
4           https://www.facebook.com/arsvivaBCN/         NaN  

[5 rows x 25 columns]

In [17]:
df_entidades.shape

(5484, 25)

**Añadimos una columna que sea 1 si tienen presencia en redes y 0 si no**.

In [18]:
# Utilizo .notna() para verificar si cada columna tiene un valor no nulo y el operador | = OR entre las condiciones de cada columna. 
# Convierto el resultado en 0 o 1 utilizando de las codniciones de las columnas .astype(int) y lo va asignando a la celda de la columna "Presencia_redes"
df_entidades["Presencia_redes"] = (df_entidades["XXSS_Instagram"].notna() | 
                                df_entidades["XXSS_Twitter"].notna() |
                                df_entidades["XXSS_Facebook"].notna() |
                                df_entidades["XXSS_Altres"].notna()).astype(int) 
df_entidades.head()

Tipologia_E_GF        Nif  Total_Voluntaris  Total_Professionals  \
0              E  G66024894               NaN                  NaN   
1              E  R0800530H               NaN                  NaN   
2              E  G08489528               NaN                  NaN   
3              E  G01675180               NaN                  NaN   
4              E  F67141648               NaN                  NaN   

  Tipus_Activitat Forma_Juridica    Codi_E_GF  \
0             NaN     Associació  99888005242   
1             NaN     Associació  99746315497   
2             NaN       Fundació  99739953046   
3             NaN     Associació  99674314027   
4             NaN    Cooperativa  99564110147   

                                    Nom_E_GF Nom_Popular Nom_Poblacio  ...  \
0  Associació d'Afectats d'Urticària Crònica         NaN    BARCELONA  ...   
1           Parròquia Sant Ignasi de Loiola          NaN    BARCELONA  ...   
2        Fundació Privada Hospital Evangèlic         NaN    BARCELONA  ...   
3     Associació d'Entreteniment Grupo Dance         NaN    BARCELONA  ...   
4                              Ars Viva SCCL         NaN    BARCELONA  ...   

   Sector_Territorial Total_Socis     Sector_Tematic_1  \
0           barcelona         NaN                Salut   
1           barcelona         NaN           Religioses   
2           barcelona         NaN                Salut   
3                 NaN       110.0              Veïnals   
4           barcelona         NaN  Educació i formació   

                   Sector_Tematic_2 Sector_Tematic_3  \
0                               NaN              NaN   
1                               NaN              NaN   
2                               NaN              NaN   
3  Promoció de la ciutadania activa    Suport social   
4                               NaN              NaN   

                          XXSS_Instagram                   XXSS_Twitter  \
0                                    NaN                            NaN   
1                                    NaN                            NaN   
2                                    NaN  https://twitter.com/NouHE_BCN   
3                                    NaN                            NaN   
4  https://www.instagram.com/arsvivabcn/                            NaN   

                                   XXSS_Facebook XXSS_Altres Presencia_redes  
0                                            NaN         NaN               0  
1                                            NaN         NaN               0  
2  https://www.facebook.com/Nouhospitalevangelic         NaN               1  
3                                            NaN         NaN               0  
4           https://www.facebook.com/arsvivaBCN/         NaN               1  

[5 rows x 26 columns]

In [19]:
df_entidades.shape

(5484, 26)

### c. Merge de las bases de datos

**Ahora voy a hacer un merge de los DataFrame como en SQL usando "outer"** para:  
- tener todas las entidades de subvencionadas con las columnas de df_entidades
- y además tener las entidades no subvencionadas con ceros en las columnas que les corresponderían.

In [20]:
# Realizo la fusión (merge) en función de la columna que contiene la clave 'Codi_E_GF' 
df_merged = pd.merge(df_subvenciones, df_entidades, on='Codi_E_GF', how='outer')
df_merged.head()

Tipologia_E_GF_x  Codi_E_GF        Entitat_Municipal  \
0                E        224  Ajuntament de Barcelona   
1                E        224  Ajuntament de Barcelona   
2                E        224  Ajuntament de Barcelona   
3                E        224  Ajuntament de Barcelona   
4                E        224  Ajuntament de Barcelona   

                                        Organ_Gestor     Any  \
0   Gerència de Política Econòmica i Desenvolupament  2019.0   
1   Gerència de Política Econòmica i Desenvolupament  2019.0   
2  Gerència d’Àrea d’Economia, Recursos i Promoci...  2022.0   
3  Gerència d’Àrea d’Economia, Recursos i Promoci...  2021.0   
4  Gerència d’Àrea d’Economia, Recursos i Promoci...  2020.0   

                              Tipologia_De_Subvencio  \
0                            SUBVENCIONS NOMINATIVES   
1  Convocatòria Subvencions Promoció i Reforç Eco...   
2  Convocatòria 2022 Impuls socioeconòmic del ter...   
3  Convocatòria 2021Promoció i Reforç de l'Econom...   
4  Convocatòria 2020 Promoció i Reforç de l'Econo...   

                                             Objecte  Import_Sollicitat  \
0        POSADA EN MARXA DEL PROJECTE SOCIAL BICIHUB           120000.0   
1                              Comuniquem el BiciHub             8000.0   
2                   L'estiu Cooperatiu de Sant Martí            40000.0   
3                                Emfortim el Bicihub            50000.0   
4  Bicicargo a Sant Martí, repartiment comunitari...            18000.0   

   Import_Total_Projecte  Import_Atorgat_Inicial  ...  Sector_Territorial  \
0               120000.0                120000.0  ...           barcelona   
1                10000.0                  4000.0  ...           barcelona   
2                50000.0                 18000.0  ...           barcelona   
3                62500.0                 32500.0  ...           barcelona   
4                29600.0                 14400.0  ...           barcelona   

  Total_Socis Sector_Tematic_1              Sector_Tematic_2  \
0        18.0  Economia social  Ecologistes i mediambientals   
1        18.0  Economia social  Ecologistes i mediambientals   
2        18.0  Economia social  Ecologistes i mediambientals   
3        18.0  Economia social  Ecologistes i mediambientals   
4        18.0  Economia social  Ecologistes i mediambientals   

         Sector_Tematic_3                                XXSS_Instagram  \
0  Entitats de segon grau  https://www.instagram.com/bicihub.barcelona/   
1  Entitats de segon grau  https://www.instagram.com/bicihub.barcelona/   
2  Entitats de segon grau  https://www.instagram.com/bicihub.barcelona/   
3  Entitats de segon grau  https://www.instagram.com/bicihub.barcelona/   
4  Entitats de segon grau  https://www.instagram.com/bicihub.barcelona/   

                   XXSS_Twitter  \
0  https://twitter.com/HubBici/   
1  https://twitter.com/HubBici/   
2  https://twitter.com/HubBici/   
3  https://twitter.com/HubBici/   
4  https://twitter.com/HubBici/   

                                       XXSS_Facebook XXSS_Altres  \
0  https://www.facebook.com/people/BiciHub-Barcel...         NaN   
1  https://www.facebook.com/people/BiciHub-Barcel...         NaN   
2  https://www.facebook.com/people/BiciHub-Barcel...         NaN   
3  https://www.facebook.com/people/BiciHub-Barcel...         NaN   
4  https://www.facebook.com/people/BiciHub-Barcel...         NaN   

  Presencia_redes  
0             1.0  
1             1.0  
2             1.0  
3             1.0  
4             1.0  

[5 rows x 36 columns]

In [21]:
df_merged.shape

(58560, 36)

In [22]:
df_merged.describe()

Codi_E_GF           Any  Import_Sollicitat  Import_Total_Projecte  \
count  5.856000e+04  56917.000000       5.543700e+04           5.655800e+04   
mean   8.499931e+10   2017.523323       1.338104e+04           2.983565e+04   
std    2.848553e+10      3.110568       7.219469e+04           1.129866e+05   
min    8.600000e+01   2011.000000       6.000000e-01           6.000000e-01   
25%    9.208601e+10   2015.000000       2.000000e+03           4.350000e+03   
50%    9.702717e+10   2018.000000       4.500000e+03           1.025000e+04   
75%    9.940024e+10   2020.000000       1.000000e+04           2.564520e+04   
max    9.988801e+10   2022.000000       7.697740e+06           1.224850e+07   

       Import_Atorgat_Inicial  Import_Reintegrat_Total  Total_Voluntaris  \
count            5.691700e+04              1485.000000         92.000000   
mean             1.222339e+04              3031.940343         27.173913   
std              1.114916e+05              9078.194119          8.456233   
min             -2.331960e+05                 0.100000          4.000000   
25%              1.000000e+03               500.000000         20.000000   
50%              2.000000e+03              1000.000000         30.000000   
75%              5.400000e+03              2500.000000         32.000000   
max              7.958646e+06            156886.000000         50.000000   

       Total_Professionals  Codi_Districte    Codi_Barri  Any_Constitucio  \
count           161.000000    58473.000000  58473.000000     58473.000000   
mean             68.832298        5.058762     28.900056      1833.636499   
std              77.436929        3.159416     23.070603       195.549459   
min               1.000000        1.000000      1.000000      1380.000000   
25%               1.000000        2.000000      8.000000      1600.000000   
50%               4.000000        5.000000     23.000000      1984.000000   
75%             170.000000        8.000000     48.000000      2003.000000   
max             170.000000       10.000000     73.000000      2023.000000   

         Total_Socis  Presencia_redes  
count   14254.000000     58473.000000  
mean     1948.898344         0.369384  
std     12562.629481         0.482642  
min         2.000000         0.000000  
25%        36.000000         0.000000  
50%       144.000000         0.000000  
75%       400.000000         1.000000  
max    300000.000000         1.000000

## 1) Análsisis de entidades

Para este anaisis usaremos el DataFrame df_entidades generado.

In [23]:
df_entidades.head()

Tipologia_E_GF        Nif  Total_Voluntaris  Total_Professionals  \
0              E  G66024894               NaN                  NaN   
1              E  R0800530H               NaN                  NaN   
2              E  G08489528               NaN                  NaN   
3              E  G01675180               NaN                  NaN   
4              E  F67141648               NaN                  NaN   

  Tipus_Activitat Forma_Juridica    Codi_E_GF  \
0             NaN     Associació  99888005242   
1             NaN     Associació  99746315497   
2             NaN       Fundació  99739953046   
3             NaN     Associació  99674314027   
4             NaN    Cooperativa  99564110147   

                                    Nom_E_GF Nom_Popular Nom_Poblacio  ...  \
0  Associació d'Afectats d'Urticària Crònica         NaN    BARCELONA  ...   
1           Parròquia Sant Ignasi de Loiola          NaN    BARCELONA  ...   
2        Fundació Privada Hospital Evangèlic         NaN    BARCELONA  ...   
3     Associació d'Entreteniment Grupo Dance         NaN    BARCELONA  ...   
4                              Ars Viva SCCL         NaN    BARCELONA  ...   

   Sector_Territorial Total_Socis     Sector_Tematic_1  \
0           barcelona         NaN                Salut   
1           barcelona         NaN           Religioses   
2           barcelona         NaN                Salut   
3                 NaN       110.0              Veïnals   
4           barcelona         NaN  Educació i formació   

                   Sector_Tematic_2 Sector_Tematic_3  \
0                               NaN              NaN   
1                               NaN              NaN   
2                               NaN              NaN   
3  Promoció de la ciutadania activa    Suport social   
4                               NaN              NaN   

                          XXSS_Instagram                   XXSS_Twitter  \
0                                    NaN                            NaN   
1                                    NaN                            NaN   
2                                    NaN  https://twitter.com/NouHE_BCN   
3                                    NaN                            NaN   
4  https://www.instagram.com/arsvivabcn/                            NaN   

                                   XXSS_Facebook XXSS_Altres Presencia_redes  
0                                            NaN         NaN               0  
1                                            NaN         NaN               0  
2  https://www.facebook.com/Nouhospitalevangelic         NaN               1  
3                                            NaN         NaN               0  
4           https://www.facebook.com/arsvivaBCN/         NaN               1  

[5 rows x 26 columns]

In [24]:
df_entidades.isna().sum()

Tipologia_E_GF            0
Nif                       3
Total_Voluntaris       5474
Total_Professionals    5475
Tipus_Activitat        5481
Forma_Juridica            3
Codi_E_GF                 0
Nom_E_GF                  0
Nom_Popular            5473
Nom_Poblacio              0
Codi_Districte            0
Nom_Districte             0
Codi_Barri                0
Nom_Barri                 0
Url_Preferent          2546
Any_Constitucio           0
Sector_Territorial     1510
Total_Socis            4148
Sector_Tematic_1          1
Sector_Tematic_2       3894
Sector_Tematic_3       4829
XXSS_Instagram         5159
XXSS_Twitter           4708
XXSS_Facebook          4385
XXSS_Altres            5193
Presencia_redes           0
dtype: int64

En un primer analisis:
- Vemos que solo hay 3 con el Tipus_Activitat distinto de Nan, y 11 con el Nom_Popular.  
- Vemos tambien que solo 1 no tiene Sector Tematic 1 (verificaré cual es) y por qué.  
- Vemos que hay un porcentaje pequeño de entidades con presencia en redes, pero lo verificaré en mas detalle.

**Tipus_Activitat**

In [25]:
df_entidades["Tipus_Activitat"].value_counts()

permanent    3
Name: Tipus_Activitat, dtype: int64

Entendiendo que donde es Nan es que es temporal, solo 3 entidades tienen actividad permanente, siendo el resto actividades temporales.

In [26]:
df_entidades["Tipologia_E_GF"].value_counts()

E     5481
GF       3
Name: Tipologia_E_GF, dtype: int64

Vemos que tenemos tambien 3 entidades que cumplen ser "Grupo de hecho" y no "entidades", y probablemente son las mismas que tienen actividad permanente.

Veamoslo:

In [27]:
df_entidades_actividad_permanente = df_entidades.loc[df_entidades["Tipus_Activitat"] == "permanent"]
df_entidades_actividad_permanente

Tipologia_E_GF  Nif  Total_Voluntaris  Total_Professionals  \
5481             GF  NaN               NaN                  NaN   
5482             GF  NaN               NaN                  NaN   
5483             GF  NaN               NaN                  NaN   

     Tipus_Activitat Forma_Juridica  Codi_E_GF  \
5481       permanent            NaN         89   
5482       permanent            NaN         87   
5483       permanent            NaN         86   

                                             Nom_E_GF Nom_Popular  \
5481                                   Dragons Gaming         NaN   
5482  Plataforma Ciutadana Millorem El Camp de l'Arpa         NaN   
5483                          C.A. Salvem El Guinardó         NaN   

     Nom_Poblacio  ...  Sector_Territorial Total_Socis  Sector_Tematic_1  \
5481    Barcelona  ...           barcelona        20.0        Esportives   
5482    Barcelona  ...  specific_districts        12.0           Veïnals   
5483    Barcelona  ...  specific_districts       128.0     Suport social   

     Sector_Tematic_2 Sector_Tematic_3  \
5481              NaN              NaN   
5482              NaN              NaN   
5483              NaN              NaN   

                                   XXSS_Instagram  \
5481   https://www.instagram.com/dragonsgamingec/   
5482                                          NaN   
5483  https://www.instagram.com/salvemelguinardo/   

                             XXSS_Twitter  \
5481  https://twitter.com/DragonsGamingec   
5482                                  NaN   
5483   https://twitter.com/salvemguinardo   

                                   XXSS_Facebook XXSS_Altres Presencia_redes  
5481                                         NaN         NaN               1  
5482                                         NaN         NaN               0  
5483  https://www.facebook.com/Salvemelguinardo/         NaN               1  

[3 rows x 26 columns]

Vemos que hay tres organizaciones con tipo de actividad "Pernamente" que no se cosndieran "Entidades" sino "Grupos de hecho", siendo el resto por omisión, Entidades temporales.  
Sería bueno verificar qué significa, es decir, si el resto se han creado ad-hoc para los proyectos o si trabajan por periodos, dado que se refiere a entidades y no a personas.

**Forma_Juridica**

In [28]:
df_entidades["Forma_Juridica"].value_counts()

Associació                 4728
Fundació                    383
Altres                      178
Cooperativa                 141
Col·legis professionals      19
Partit / Sindicat            17
Gremis                       15
Name: Forma_Juridica, dtype: int64

Entendiendo que donde es Nan es que es temporal, solo 3 entidades tienen actividad permanente, siendo el resto actividades temporales.

In [29]:
df_entidades["Forma_Juridica"].value_counts()

Associació                 4728
Fundació                    383
Altres                      178
Cooperativa                 141
Col·legis professionals      19
Partit / Sindicat            17
Gremis                       15
Name: Forma_Juridica, dtype: int64

Vemos que tenemos tambien 3 entidades que cumplen ser "Grupo de hecho" y no "entidades", y probablemente son las mismas que tienen actividad permanente.

Veamoslo:

In [30]:
df_entidades_actividad_permanente = df_entidades.loc[df_entidades["Forma_Juridica"] == "permanent"]
df_entidades_actividad_permanente

Empty DataFrame
Columns: [Tipologia_E_GF, Nif, Total_Voluntaris, Total_Professionals, Tipus_Activitat, Forma_Juridica, Codi_E_GF, Nom_E_GF, Nom_Popular, Nom_Poblacio, Codi_Districte, Nom_Districte, Codi_Barri, Nom_Barri, Url_Preferent, Any_Constitucio, Sector_Territorial, Total_Socis, Sector_Tematic_1, Sector_Tematic_2, Sector_Tematic_3, XXSS_Instagram, XXSS_Twitter, XXSS_Facebook, XXSS_Altres, Presencia_redes]
Index: []

[0 rows x 26 columns]

Vemos que hay tres organizaciones con tipo de actividad "Pernamente" que no se cosndieran "Entidades" sino "Grupos de hecho", siendo el resto por omisión, Entidades temporales.  
Sería bueno verificar qué significa, es decir, si el resto se han creado ad-hoc para los proyectos o si trabajan por periodos, dado que se refiere a entidades y no a personas.

**Nom_Popular**

In [31]:
df_entidades_nom_popular = df_entidades.loc[df_entidades["Nom_Popular"].notna()]
df_entidades_nom_popular

Tipologia_E_GF        Nif  Total_Voluntaris  Total_Professionals  \
2343              E  G63566467              40.0                  4.0   
3610              E  G08943813               NaN                170.0   
3720              E  G58071507              32.0                 61.0   
5410              E  G85875409              11.0                  NaN   
5433              E  F01909373               NaN                  NaN   
5438              E  G44965960               NaN                  NaN   
5443              E  G66712696               NaN                  NaN   
5459              E  G59182766               NaN                  NaN   
5460              E  G10972099               NaN                  NaN   
5461              E  G65506917              35.0                  NaN   
5474              E  G24815664              50.0                  5.0   

     Tipus_Activitat Forma_Juridica    Codi_E_GF  \
2343             NaN     Associació  99400172854   
3610             NaN     Associació  92086022186   
3720             NaN       Fundació  92086015086   
5410             NaN     Associació          299   
5433             NaN    Cooperativa          276   
5438             NaN     Associació          270   
5443             NaN     Associació          265   
5459             NaN     Associació          247   
5460             NaN     Associació          246   
5461             NaN     Associació          245   
5474             NaN     Associació          227   

                                               Nom_E_GF  \
2343  Federació d'Associacions Catalanes de Pares i ...   
3610                    ASSOCIACIO ESPORTIVA L'EIXAMPLE   
3720  La Tutela, Fundació privada catalana tutelar d...   
5410  PLATAFORMA INTERNACIONAL DE COOPERACIÓN Y MIGR...   
5433                               CircularBCN S.C.C.L.   
5438  Associació VIS, Associació de persones migrade...   
5443                             Associació Gospel Soul   
5459       Associació Rotary Club de Barcelona Diagonal   
5460  Diverços, Associació de Comerciants i Empresar...   
5461  Associació pel LLoguer Públic i Assequible ( A...   
5474                           no publicar test canviat   

                                            Nom_Popular Nom_Poblacio  ...  \
2343  Federació ACAPPS de Famílies i Persones amb So...    BARCELONA  ...   
3610                                        AE EIXAMPLE    BARCELONA  ...   
3720                                          La Tutela    BARCELONA  ...   
5410  PLATAFORMA INTERNACIONAL DE COOPERACIÓN Y MIGR...    Barcelona  ...   
5433                                              Rebag    Barcelona  ...   
5438                 Associació VIS, Moldàvia i Romania    Barcelona  ...   
5443                                        Gospel Soul    Barcelona  ...   
5459                     Rotary club Barcelona Diagonal    Barcelona  ...   
5460                                Divers de Barcelona    Barcelona  ...   
5461                                         As. 500x20    Barcelona  ...   
5474                                              no pu    Barcelona  ...   

      Sector_Territorial Total_Socis  \
2343           barcelona    107840.0   
3610  specific_districts        32.0   
3720           barcelona         NaN   
5410           barcelona        21.0   
5433           barcelona         2.0   
5438           barcelona        50.0   
5443           barcelona         5.0   
5459  specific_districts        41.0   
5460           barcelona         5.0   
5461  specific_districts        35.0   
5474           barcelona       500.0   

                                      Sector_Tematic_1  \
2343                                     Discapacitats   
3610                                        Esportives   
3720                                     Suport social   
5410  Cooperació al desenvolupament, Solidaritat i Pau   
5433                      Ecologistes i mediambientals   
5438  Cooperació al desenvolupament, Sol

Encontramos 11 Entidades con nombre popular.

## 2) Análisis de correlación entre entidades subveniconadas y no subvencionadas

## 2) Análsis de entidades subvencionadas

Crecimiento anula de subvenciones

**Por tipus de Activitat**

Veamos si han sido subvencionadas:

In [32]:
df_entidades_actividad_permanente_subvencionadas=df_merged.loc[df_merged["Tipus_Activitat"] == "permanent"]
df_entidades_actividad_permanente_subvencionadas

Tipologia_E_GF_x  Codi_E_GF Entitat_Municipal Organ_Gestor  Any  \
58557              NaN         89               NaN          NaN  NaN   
58558              NaN         87               NaN          NaN  NaN   
58559              NaN         86               NaN          NaN  NaN   

      Tipologia_De_Subvencio Objecte  Import_Sollicitat  \
58557                    NaN     NaN                NaN   
58558                    NaN     NaN                NaN   
58559                    NaN     NaN                NaN   

       Import_Total_Projecte  Import_Atorgat_Inicial  ...  Sector_Territorial  \
58557                    NaN                     NaN  ...           barcelona   
58558                    NaN                     NaN  ...  specific_districts   
58559                    NaN                     NaN  ...  specific_districts   

      Total_Socis Sector_Tematic_1  Sector_Tematic_2  Sector_Tematic_3  \
58557        20.0       Esportives               NaN               NaN   
58558        12.0          Veïnals               NaN               NaN   
58559       128.0    Suport social               NaN               NaN   

                                    XXSS_Instagram  \
58557   https://www.instagram.com/dragonsgamingec/   
58558                                          NaN   
58559  https://www.instagram.com/salvemelguinardo/   

                              XXSS_Twitter  \
58557  https://twitter.com/DragonsGamingec   
58558                                  NaN   
58559   https://twitter.com/salvemguinardo   

                                    XXSS_Facebook XXSS_Altres Presencia_redes  
58557                                         NaN         NaN             1.0  
58558                                         NaN         NaN             0.0  
58559  https://www.facebook.com/Salvemelguinardo/         NaN             1.0  

[3 rows x 36 columns]

No han sido subvencionadas.

**Por Nom popular**

In [33]:
df_entidades_nom_popular_subvencionadas=df_merged.loc[df_merged["Nom_Popular"].notna()]
df_entidades_nom_popular_subvencionadas

Tipologia_E_GF_x    Codi_E_GF                     Entitat_Municipal  \
3922                 E  92086022186  Institut Municipal Barcelona Esports   
3923                 E  92086022186  Institut Municipal Barcelona Esports   
3924                 E  92086022186  Institut Municipal Barcelona Esports   
3925                 E  92086022186  Institut Municipal Barcelona Esports   
3926                 E  92086022186  Institut Municipal Barcelona Esports   
...                ...          ...                                   ...   
58530              NaN          265                                   NaN   
58540              NaN          247                                   NaN   
58541              NaN          246                                   NaN   
58542              NaN          245                                   NaN   
58552              NaN          227                                   NaN   

      Organ_Gestor     Any                      Tipologia_De_Subvencio  \
3922           NaN  2017.0  SUBVENCIONS DIRECTES (amb o sense CONVENI)   
3923           NaN  2021.0  SUBVENCIONS DIRECTES (amb o sense CONVENI)   
3924           NaN  2021.0                                       AJUTS   
3925           NaN  2021.0                                       AJUTS   
3926           NaN  2019.0  SUBVENCIONS DIRECTES (amb o sense CONVENI)   
...            ...     ...                                         ...   
58530          NaN     NaN                                         NaN   
58540          NaN     NaN                                         NaN   
58541          NaN     NaN                                         NaN   
58542          NaN     NaN                                         NaN   
58552          NaN     NaN                                         NaN   

                              Objecte  Import_Sollicitat  \
3922    Ajuts 2017-2018 (1r pagament)                NaN   
3923                  Ajuts 2020-2021           32932.51   
3924                Ajut del programa           44435.19   
3925                Ajut del programa            7163.40   
3926   Ajuts 2018-2019 (2n. pagament)           12855.56   
...                               ...                ...   
58530                             NaN                NaN   
58540                             NaN                NaN   
58541                             NaN                NaN   
58542                             NaN                NaN   
58552                             NaN                NaN   

       Import_Total_Projecte  Import_Atorgat_Inicial  ...  Sector_Territorial  \
3922                43509.42                43509.42  ...  specific_districts   
3923                32932.51                  464.83  ...  specific_districts   
3924                44435.19                33326.95  ...  specific_districts   
3925                 7163.40                 5372.62  ...  specific_districts   
3926                12855.56                12855.56  ...  specific_districts   
...                      ...                     ...  ...                 ...   
58530                    NaN                     NaN  ...           barcelona   
58540                    NaN                     NaN  ...  specific_districts   
58541                    NaN                     NaN  ...           barcelona   
58542                    NaN                     NaN  ...  specific_districts   
58552                    NaN                     NaN  ...           barcelona   

      Total_Socis                  Sector_Tematic_1        Sector_Tematic_2  \
3922         32.0                        Esportives               Gent gran   
3923         32.0                        Esportives               Gent gran   
3924         32.0                        Esportives               Gent gran   
3925         32.0                        Esportives               Gent gran   
3926         32.0                        Esportives               Gent gran   
...           ...       

In [34]:
suma_importes = df_entidades_nom_popular_subvencionadas["Import_Atorgat_Inicial"].sum()
print(f"La suma de la columna Import_Atorgat_Inicial es: {suma_importes}")

La suma de la columna Import_Atorgat_Inicial es: 1481039.23


Presentació del conjunt de dades escollit: explicació i observacions generals del conjunt de dades a utilitzar per a fer el projecte final.
Característiques generals: sense necessitat d'entrar molt en detall, explicar les principals característiques que defineixen aquest conjunt de dades. Tipologia, sector, tipus de dades, font, context, etc.
Definició de les variables: explicació teòrica de les principals variables que conté el conjunt de dades.
Presentació dels objectius: detallar els objectius inicials marcats de cara a extreure informació rellevant del conjunt de dades.